caom2-pkg-server/src/main/java/ca/nrc/cadc/caom2/pkg/Caom2PackageRunner.java

```diff
@@ -0,0 +1,272 @@
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+*  $Revision: 5 $
+*
+************************************************************************
+*/
+
+package ca.nrc.cadc.caom2.pkg;
+
+import org.opencadc.pkg.server.PackageItem;
+import org.opencadc.pkg.server.PackageRunner;
+import org.opencadc.pkg.server.PackageRunnerException;
+
+import ca.nrc.cadc.auth.AuthMethod;
+import ca.nrc.cadc.auth.AuthenticationUtil;
+import ca.nrc.cadc.caom2.Artifact;
+import ca.nrc.cadc.caom2.ObservationURI;
+import ca.nrc.cadc.caom2.PlaneURI;
+import ca.nrc.cadc.caom2.ProductType;
+import ca.nrc.cadc.caom2.PublisherID;
+import ca.nrc.cadc.caom2.artifact.resolvers.CaomArtifactResolver;
+import ca.nrc.cadc.caom2ops.ArtifactQueryResult;
+import ca.nrc.cadc.caom2ops.CaomTapQuery;
+import ca.nrc.cadc.caom2ops.ServiceConfig;
+import ca.nrc.cadc.log.WebServiceLogInfo;
+import ca.nrc.cadc.net.ResourceNotFoundException;
+import ca.nrc.cadc.util.StringUtil;
+import ca.nrc.cadc.uws.ErrorSummary;
+import ca.nrc.cadc.uws.ErrorType;
+import ca.nrc.cadc.uws.ParameterUtil;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URISyntaxException;
+import java.security.AccessControlContext;
+import java.security.AccessController;
+import java.security.cert.CertificateException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+import javax.security.auth.Subject;
+import org.apache.log4j.Logger;
+
+
+public class Caom2PackageRunner extends PackageRunner {
+    private static final Logger log = Logger.getLogger(Caom2PackageRunner.class);
+
+    private final URI tapID;
+    private List<String> idList;
+    private String packageName;
+
+    public Caom2PackageRunner()
+    { 
+        ServiceConfig sc = new ServiceConfig();
+        this.tapID = sc.getTapServiceID();
+    }
+
+    @Override
+    protected String getPackageName() {
+        return this.packageName;
+    }
+
+
+    @Override
+     public Iterator<PackageItem> getItems() throws IOException, PackageRunnerException {
```

Comment:
I was curious what `PackageRunnerException` was going to be used for and it turns out it isn't used at all. Although the existing error handling in this method is lacking :blush: this doesn't really add anything... it does reinforce that having a separate "init" method where most failures will occur would help.

Reply:
PackageRunnerException - at one point I had code in getItems that needed exceptions thrown or caught, and I wanted a way to percolate them out to the sendError functions in the base PackageRunner code so they could be reported and not lost in logs.  The current implementation of (now called CaomPackageRunner??) - didn't end up using it.

Reply:
Almost used it in initPackage, but everything was throwing an IllegalArgumentException so opted for that instead.